script.js

```diff
@@ -1,4 +1,110 @@
+var computerhand = ``;
+var winCount = 0;
+var roundCount = 0;
+var gameMode = `waiting for username`;
+var userName = "";
+var playerHand = ``;
+
+// ======================================MAIN FUNCTION STARTS BELOW ====================================
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  myOutputValue = ``;
+  //Username needed to start the game
+  if (gameMode == `waiting for username`) {
+    userName = input;
+    gameMode = `SPS`;
+    myOutputValue = `Hello ${userName}! <br> <br> You are now ready to play. <br> <br> Paper, scissors, or stone?`;
+  } else if (gameMode == `SPS`) {
+    myOutputValue = spsGame(input);
+  }
+
+  return `${myOutputValue} <br> <br> GAME MODE: ${gameMode} <br> <br> Rounds played: ${roundCount}`;
+};
+
+// ===============================MAIN FUNCITON ENDS HERE==================================================
+
+// This function is to compare hands and check whether player or computer won
+var WinnerChecker = function (playerHand, computerhand) {
+  // THIS FUNCTION IS FOR PLAYER WINS
+  if (
+    (playerHand == `scissors` && computerhand == `paper`) ||
+    (playerHand == `paper` && computerhand == `stone`) ||
+    (playerHand == `stone` && computerhand == `scissors`)
+  ) {
+    return `Player win`;
+  }
+
+  // THIS FUCTION IS FOR TIE
+  if (
+    (playerHand == `scissors` && computerhand == `scissors`) ||
+    (playerHand == `paper` && computerhand == `paper`) ||
+    (playerHand == `stone` && computerhand == `stone`)
+  ) {
+    return `Tie`;
+  }
+
+  // THIS IS FOR PLAYER LOSES
+  if (
+    (playerHand == `scissors` && computerhand == `stone`) ||
+    (playerHand == `paper` && computerhand == `scissors`) ||
+    (playerHand == `stone` && computerhand == `paper`)
+  ) {
+    return `Player loses`;
+  }
+};
+
+// SPS Game function
+var spsGame = function (playerHand) {
+  var msg = ``;
+
+  randomnumber = generaterandomnumber();
+
+  // connect each number to either SPS
+  if (randomnumber == 1) {
+    var computerhand = "scissors";
+  }
+  if (randomnumber == 2) {
+    var computerhand = "paper";
+  }
+  if (randomnumber == 3) {
+    var computerhand = "stone";
+  }
+
+  var WinnerResult = WinnerChecker(playerHand, computerhand);
+
+  // This is what the player sees if they lose against the computer
+  if (WinnerResult == `Player loses`) {
+    roundCount = roundCount + 1;
+    var msg = `${userName} you lose! You showed ${playerHand} and the computer showed ${computerhand}. Your total wins are ${winCount}.`;
```

Comment:
you have already declared a variable `msg` in line 58, so you don't need to re-declare it here, but you can just replace it's value. meaning you can just say
`msg = ${userName} you lose! You showed ${playerHand} and the computer showed ${computerhand}. Your total wins are ${winCount}.;`